R/grid_search.R

```diff
@@ -0,0 +1,124 @@
+##' Run a grid search of the particle filter over beta and start date
+##' 
+##' @title Grid search of beta and start date
+##' 
+##' @param min_beta Minimum value of beta in the search
+##' 
+##' @param max_beta Maximum value of beta in the search
+##' 
+##' @param beta_step Step to increment beta between min and max
+##' 
+##' @param first_start_date Earliest start date as 'yyyy-mm-dd'
+##' 
+##' @param last_start_date Latest start date as 'yyyy-mm-dd'
+##' 
+##' @param day_step Step to increment date in days
+##' 
+##' @param data Hosptial data to fit to. See \code{example.csv}
+##'   and \code{particle_filter_data()}
+##' 
+##' @param model_params Model parameters, from a call to
+##'   \code{generate_parameters()}. If NULL, uses defaults as
+##'   in unit tests.
+##'   
+##' @param pars_obs list of parameters to use in comparison
+##'   with \code{compare_icu}.
+##'   
+##' @return List of beta and start date grid values, and
+##'   normalised probabilities at each point
+##' 
+##' @export
+##' @import furrr
+##' @import future
+scan_beta_date <- function(
```

Comment:
OK, I think we can generalise this here: how about:

```
scan_likelihood <- function(pars, data, model_params, pars_obs) {
  param_grid <- do.call("expand.grid", pars)
  ...
}
```

Then we need to generalise the approach below.  Or we can proceed with this and I can pop that generalisation together later and get you to review?